### DISCUSSION TITLE
Task extension does not work

### DISCUSSION THREAD

**Discussion by emilliepost** [AntlerVC/firetable#1189] [Support / Q&A] [answered]
Hi there,

I created a new task extension (it should ref.update a row) that is triggered when I check the toggle.
<img width="737" alt="Screenshot 2023-03-30 at 18 17 20" src="https://user-images.githubusercontent.com/72376544/228900224-c31f898a-bf14-4d9e-817a-ed84f7594d0b.png">

I have the same script for my Action that works fine.

I checked the logs and there is no error.
<img width="515" alt="Screenshot 2023-03-30 at 18 17 59" src="https://user-images.githubusercontent.com/72376544/228900362-a8956d1f-5f9a-43c7-bdc8-e8cd7aa7e17b.png">

Here is the snippet with comments:

```
const extensionBody: TaskBody = async({row, db, change, ref, logging}) => {

  const chapters = row.Chapters;
  const author = row.Author;
  const title = row.Title;

  // Create an array to store the results
  const results = [];

  // Loop through each chapter object in the array
  for (let i = 0; i < chapters.length; i++) {

    // Extract the content and name properties from the current chapter object
    const { Content, Name } = chapters[i];

    // Build the message content string using the $variable syntax
    const messageContent = *some_content*`;

    // Make the API request with the message content
    const res = await fetch("https://api.openai.com/v1/chat/completions", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "Authorization": "*Bearer token*"
      },
      body: JSON.stringify({
        "model": "gpt-3.5-turbo",
        "max_tokens": 3000,
        "temperature": 0.9,
        "messages": [
          {
            "role": "user",
            "content": messageContent
          }
        ]
      })
    });

    // Extract the JSON response from the API request
    const result = await res.json();

    // Push the result to the results array
    results.push(result);

  }

  // Update the  document with the results array
  await ref.update({ listOne: results });

  return {
    message: "Model Started",
    success: true
  }
  logging.log("extensionBody finished")

}
```
Could you please advise why it does nothing when triggered? I also tried using Derivative and got the same result.

Thanks a lot 🙏

**Comment by bofeiw** [comment, marked as answer]
Looks like the log you attached is from conditions, not the function body.

Do you see this line printed in Rowy logs if you move it before the return statement?
```js
logging.log("extensionBody finished")
```

It could be helpful to add more `logging.log` to see where the code is actually doing.
